### PR TITLE
SCons: Process wildcards when passed inside list of sources

### PR DESCRIPTION
**Pull request by ghost**
Very useful for thirdparty

Converted only some thirdparty sources to wildcards because:

1. Want to look if CI fails
2. Some of thirdparty sources can't be wildcarded becasue some folders contain more files than listed in the main list of files (some are conditionally added but some can be unused at all though I haven't investigated it yet)

### CODE REVIEW

**Comment by dustdfg** [comment]
I really assume that wildcards weren't used for thirparty sources intentionally but not sure. If it is intentional please tell me. But even if it is intentional I am working on SCU for thirdparty sources and wildcards for thirdparties is one of the ways to achieve in future

File: methods.py [hunk 1]
```diff
@@ -72,22 +72,25 @@ def print_error(*values: object) -> None:
 
 
 def add_source_files_orig(self, sources, files, allow_gen=False):
-    # Convert string to list of absolute paths (including expanding wildcard)
     if isinstance(files, str):
+        files = [files]
+
+    for path in files:
+        _files = self.Glob(str(path))
```
Comment (dustdfg) [inline methods.py:79]: Here if `path` is already `env.{File/Entry}` or something like it is converted into string to be passed to glob and be again reconverted to `env.{File/Entry}`... It is bad but it works and have less lines...

**Comment by akien-mga** [comment]
It is intentional for thirdparty code that we don't use wildcards. We want to make sure we review clearly which files we're building and why, and removes the ones we don't use. So this change isn't desired.

**Comment by dustdfg** [comment]
> It is intentional for thirdparty code that we don't use wildcards. We want to make sure we review clearly which files we're building and why, and removes the ones we don't use. So this change isn't desired.

In this case I will need a way to convert those manual lists into wildcards in runtime or rewrite SCU to work not only with wildcards...

**Comment by akien-mga** [comment]
I think the SCU code should be adapted if it can't deal with an explicit list of files. CC @lawnjelly

**Comment by dustdfg** [comment]
> I think the SCU code should be adapted if it can't deal with an explicit list of files

In general it is exactly what I am doing now ;) All my recent PR's are puzzles of a big net of new *fixed* SCU which can normally work with SCons paths, wildcards, multiple dirs (already in codebase but broken), subdir merging and explicit lists. Tbh I wanted to leave explicit lists for the end and didn't expect this PR to be merged but hoped to leave for the end. OK I will reorder my plans. I think I can do everything about it myself

**Comment by dustdfg** [comment]
> I think the SCU code should be adapted if it can't deal with an explicit list of files

The reason why it works only with wildcards is very simple. If you process a folder it is very easy to generate header which includes all `*.cpp` files (which isn't so IIRC and generates SCU files more granularly). But if you allow SCU only for wildcarded paths you ensure that there is no any errors or prolems...

**Comment by dustdfg** [comment]
Was a test field. The change of allowing SCons `#` paths inside arrays is not perfect. I don't want to polish it. If you really need it you can extract it but I doubt it is so necessary

**Pull request closed without merging**